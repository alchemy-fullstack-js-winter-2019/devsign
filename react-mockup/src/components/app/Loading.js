import React from 'react';
import styled from 'styled-components';
import Footer from './Footer';
import Header from './Header';

const Loader = styled.div`
  h1 {
    font-family: 'Montserrat', sans-serif;
    position: relative;
    text-align: center;
    color: rgba(0, 0, 0, .3);
    font-size: 5em
  }
  h1:before {
      content: attr(data-text);
      position: absolute;
      overflow: hidden;
      max-width: 7em;
      white-space: nowrap;
      color: #3498db;
      animation: loading 8s linear;
  }
  @keyframes loading {
      0% {
          max-width: 0;
      }
  }
`;

function Loading() {
  return (
    <>
      <Header />
      <Loader>
        <h1 data-text="Loading…">Loading…</h1>
      </Loader>
      <Footer />
    </>
  );
}

export default Loading;
