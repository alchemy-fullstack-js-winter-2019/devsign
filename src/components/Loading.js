import React from 'react';
import styles from './css/Loading.css';

export default function Loading() {
  return (
    <section className={styles.loading}>
      <p>Loading....</p>
    </section>
  );
}
