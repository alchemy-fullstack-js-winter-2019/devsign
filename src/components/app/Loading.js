import React from 'react';

export default function Loading() {
  return (
    <body>
      <main>
        <h1>...LOADING</h1>
        <img src="../assets/logo.png" alt="logo"/>
      </main>
    </body>
  );
}
