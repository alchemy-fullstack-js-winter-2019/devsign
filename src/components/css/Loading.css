.loading {
  font-size: 30px;
  background: #fff;
  border: 10px solid #4cd7d0;
  position: absolute;
  left: 50%;
  top: 30%;
  margin: -60px 0 0 -60px;
  height: 100px;
}
.loading:after {
  content: '';
  width: 200%;
  height: 200%;
  position: absolute;
  top: -20%;
  left: -20%;
  box-shadow:#fda085  -4px -5px 3px -3px;
  animation: rotate 2s infinite linear;
}

@keyframes rotate {
  0% {
    transform: rotateZ(0deg);
  }
  100% {
    transform: rotateZ(360deg);
  }
}
