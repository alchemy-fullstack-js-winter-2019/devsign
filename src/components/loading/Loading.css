
h2 {
  text-align: center;
}
img{
  display: block;
  margin: auto;
}