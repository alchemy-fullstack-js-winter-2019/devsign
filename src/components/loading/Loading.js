import React from 'react';
import styles from './Loading.css';

export default function Loading() {
  return (
    <>
      <h2 style={styles}>TWITTER ...</h2>
      <figure>
        <img src='https://i.kinja-img.com/gawker-media/image/upload/s--2m8vKwCk--/c_scale,f_auto,fl_progressive,q_80,w_800/17oyh0zri482jjpg.jpg' alt='twitter bird logo' />
      </figure>
    </>
  );
}
