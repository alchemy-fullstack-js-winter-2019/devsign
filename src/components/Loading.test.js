import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './Loading';

describe('Loading', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Loading />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
