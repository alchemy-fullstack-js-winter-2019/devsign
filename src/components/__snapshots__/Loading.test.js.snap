// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading matches a snapshot 1`] = `
<body>
  <main>
    <h1>
      ...LOADING
    </h1>
    <img
      src="../assets/logo.png"
    />
  </main>
</body>
`;
